### PR TITLE
PRD-5865 - Pentaho Platform has inconsistent methods to connect to Mo…

### PR DESCRIPTION
**Pull request by YuryBY**
…ndrian, creating multiple caches, of which only one can be cleared.

@pamval, @tmorgner, @lucboudreau, could you please take a look?

### CODE REVIEW

**Comment by YuryBY** [comment]
@pamval, @tmorgner, @lucboudreau, the PR was updated tests pass locally.

**Comment by YuryBY** [comment]
@pamval, @tmorgner, @lucboudreau, any updates?

**Comment by YuryBY** [comment]
@pamval, @tmorgner, @lucboudreau, any updates?

File: extensions/src/main/java/org/pentaho/platform/plugin/action/olap/impl/OlapServiceImpl.java [hunk 1]
```diff
@@ -504,6 +504,12 @@ public void flushAll( IPentahoSession session ) {
       resetCache( session );
 
       flushHostedAndRemote( session );
+
+      MondrianServer mondrianServer =
```
Comment (tmorgner) [inline extensions/src/main/java/org/pentaho/platform/plugin/action/olap/impl/OlapServiceImpl.java]: You will need to clear both server instances. The bug states that there are inconsistent methods for actually acquiring a mondrian instance used by the various tools. Your patch here does not address that, it only implements the workaround outlined in the case.

As there are still multiple instances alive, you have to call the flushSchemaCache() line on both the active mondrian server (this.server) and the one with Id "null". You cant just clear one, but not the other.

**Comment by YuryBY** [comment]
@tmorgner, the PR is updated. Thanks.

**Pull request merged**